packages/core/src/render3/node_manipulation.ts

```diff
@@ -689,111 +692,46 @@ export function nativeRemoveNode(renderer: Renderer3, rNode: RNode, isHostElemen
-    ngContainerChildTNode = ngContainerChildTNode.next;
-  }
-}
 
 /**
- * Appends a projected node to the DOM, or in the case of a projected container,
- * appends the nodes from all of the container's active views to the DOM.
- *
- * @param projectedTNode The TNode to be projected
- * @param tProjectionNode The projection (ng-content) TNode
- * @param currentView Current LView
- * @param projectionView Projection view (view above current)
+ * Performs the operation of `action` on the node. Typically this involves inserting or removing
+ * nodes on the LView or projection boundary.
  */
-function appendProjectedNode(
-    projectedTNode: TNode, tProjectionNode: TNode, currentView: LView,
-    projectionView: LView): void {
-  const native = getNativeByTNode(projectedTNode, projectionView);
-  appendChild(native, tProjectionNode, currentView);
-
-  // the projected contents are processed while in the shadow view (which is the currentView)
-  // therefore we need to extract the view where the host element lives since it's the
-  // logical container of the content projected views
-  attachPatchData(native, projectionView);
-
-  const nodeOrContainer = projectionView[projectedTNode.index];
-  if (projectedTNode.type === TNodeType.Container) {
-    // The node we are adding is a container and we are adding it to an element which
-    // is not a component (no more re-projection).
-    // Alternatively a container is projected at the root of a component's template
-    // and can't be re-projected (as not content of any component).
-    // Assign the final projection location in those cases.
-    for (let i = CONTAINER_HEADER_OFFSET; i < nodeOrContainer.length; i++) {
-      addRemoveViewFromContainer(nodeOrContainer[i], true, nodeOrContainer[NATIVE]);
-    }
-  } else if (projectedTNode.type === TNodeType.IcuContainer) {
-    // The node we are adding is an ICU container which is why we also need to project all the
-    // children nodes that might have been created previously and are linked to this anchor
-    let ngContainerChildTNode: TNode|null = projectedTNode.child as TNode;
-    appendProjectedChildren(
-        ngContainerChildTNode, ngContainerChildTNode, projectionView, projectionView);
-  } else {
-    if (projectedTNode.type === TNodeType.ElementContainer) {
-      appendProjectedChildren(projectedTNode.child, tProjectionNode, currentView, projectionView);
+function applyNodes(
+    renderer: Renderer3, action: WalkTNodeTreeAction, tNode: TNode | null, lView: LView,
+    renderParent: RElement | null, beforeNode: RNode | null, isProjection: boolean) {
+  while (tNode != null) {
+    ngDevMode && assertTNodeForLView(tNode, lView);
+    ngDevMode && assertNodeOfPossibleTypes(
+                     tNode, TNodeType.Container, TNodeType.Element, TNodeType.ElementContainer,
+                     TNodeType.Projection, TNodeType.Projection, TNodeType.IcuContainer);
+    const nativeNode = lView[tNode.index];
```

Comment:
Since this value can be a component LView or LContainer, seems like we should rename this var?
```suggestion
    const rawSlotValue = lView[tNode.index];
```